contracts/src/v0.8/automation/dev/test/AutomationRegistry2_3.t.sol

```diff
@@ -546,8 +577,15 @@ contract SetConfig is SetUp {
       offchainConfigBytes
     );
 
-    (, , address[] memory signers, address[] memory transmitters, uint8 f) = registry.getState();
+    (
+      ,
+      IAutomationV21PlusCommon.OnchainConfigLegacy memory onchainConfig2,
+      address[] memory signers,
+      address[] memory transmitters,
+      uint8 f
+    ) = registry.getState();
 
+    assertEq(onchainConfig2.registrars.length, 3);
```

Comment:
Can we test the values of the new registrars, instead of just the length? Also, I think this bug is only exposed when _decreasing_ registrars. Could we extend this test so that we decrease from say 3 to 0 and test that all have been removed?

Reply:
> Can we test the values of the new registrars, instead of just the length? 

sure, will do.

> I think this bug is only exposed when decreasing registrars. Could we extend this test so that we decrease from say 3 to 0 and test that all have been removed?

This should be already covered. When we setConfig, we remove all existing registrars and set new ones. If you check out this particular commit, and run the foundry test, you should see the test failure. I left it there in a separate commit to showcase this.

https://github.com/smartcontractkit/chainlink/pull/12885/commits/ff05527737887a5e9a71fe5e48a118e4c81ceb40